Mathlib/Probability/Moments/IntegrableExpMul.lean

```diff
@@ -0,0 +1,331 @@
+
+/-- If `rexp ((v + t) * X)` and `rexp ((v - t) * X)` are integrable, then for all `n : ℕ`,
+`|X| ^ n * exp (v * X)` is integrable. -/
+lemma integrable_pow_abs_mul_exp_of_integrable_exp_mul (ht : t ≠ 0)
+    (ht_int_pos : Integrable (fun ω ↦ rexp ((v + t) * X ω)) μ)
+    (ht_int_neg : Integrable (fun ω ↦ rexp ((v - t) * X ω)) μ) (n : ℕ) :
+    Integrable (fun ω ↦ |X ω| ^ n * exp (v * X ω)) μ := by
+  suffices Integrable (fun ω ↦ (t * |X ω|) ^ n / n.factorial * exp (v * X ω)) μ by
+    have h_eq ω : |X ω| ^ n * exp (v * X ω)
+        = ((t * |X ω|) ^ n / n.factorial * exp (v * X ω)) * n.factorial / t ^ n := by
+      rw [mul_pow]
+      field_simp
+      ring
+    simp_rw [h_eq]
+    exact (this.mul_const _).div_const _
+  have h_le ω : (|t| * |X ω|) ^ n / n.factorial ≤ exp (|t| * |X ω|) :=
+    pow_div_factorial_le_exp _ (by positivity) _
+  have h_int := integrable_exp_abs_mul_abs_add ht_int_pos ht_int_neg
+  refine Integrable.mono h_int ?_ (ae_of_all _ fun ω ↦ ?_)
+  · have hX : AEMeasurable X μ := by
+      by_cases hvt : v + t = 0
+      · have hvt' : v - t ≠ 0 := by
+          rw [sub_ne_zero]
+          refine fun h_eq ↦ ht ?_
+          simpa [h_eq] using hvt
+        exact aemeasurable_of_aemeasurable_exp_mul hvt' ht_int_neg.aemeasurable
+      · exact aemeasurable_of_aemeasurable_exp_mul hvt ht_int_pos.aemeasurable
+    simp_rw [mul_pow]
+    refine AEMeasurable.aestronglyMeasurable ?_
+    exact (((hX.abs.pow_const _).const_mul _).div_const _).mul
+      (measurable_exp.comp_aemeasurable (hX.const_mul _))
+  · simp only [norm_div, norm_pow, norm_mul, norm_eq_abs, abs_abs, norm_natCast, abs_exp,
+      Nat.abs_cast]
+    rw [exp_add]
+    gcongr
+    exact h_le _
+
+/-- If `rexp ((v + t) * X)` and `rexp ((v - t) * X)` are integrable, then for all `n : ℕ`,
+`X ^ n * exp (v * X)` is integrable. -/
+lemma integrable_pow_mul_exp_of_integrable_exp_mul (ht : t ≠ 0)
+    (ht_int_pos : Integrable (fun ω ↦ rexp ((v + t) * X ω)) μ)
+    (ht_int_neg : Integrable (fun ω ↦ rexp ((v - t) * X ω)) μ) (n : ℕ) :
+    Integrable (fun ω ↦ X ω ^ n * exp (v * X ω)) μ := by
+  rw [← integrable_norm_iff]
+  · simp_rw [norm_eq_abs, abs_mul, abs_pow, abs_exp]
+    exact integrable_pow_abs_mul_exp_of_integrable_exp_mul ht ht_int_pos ht_int_neg n
+  · have hX : AEMeasurable X μ := by
+      by_cases hvt : v + t = 0
+      · have hvt' : v - t ≠ 0 := by
+          rw [sub_ne_zero]
+          refine fun h_eq ↦ ht ?_
+          simpa [h_eq] using hvt
+        exact aemeasurable_of_aemeasurable_exp_mul hvt' ht_int_neg.aemeasurable
+      · exact aemeasurable_of_aemeasurable_exp_mul hvt ht_int_pos.aemeasurable
+    exact ((hX.pow_const _).mul
+      (measurable_exp.comp_aemeasurable (hX.const_mul _))).aestronglyMeasurable
+
+/-- If `ω ↦ rexp (t * X ω)` is integrable at `t` and `-t` for `t ≠ 0`, then `ω ↦ |X ω| ^ n` is
+integrable for all `n : ℕ`. That is, all moments of `X` are finite. -/
+lemma integrable_pow_abs_of_integrable_exp_mul (ht : t ≠ 0)
```

Comment:
Is it easy to also derive a `memLp` statement from this one or the next one (for real exponents, not only integers ones)?